AlphaWallet/Core/UnstoppableDomains/DomainResolutionService.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  DomainResolutionServiceType.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 27.01.2022.
+//
+
+import Foundation
+import PromiseKit
+
+class DomainResolutionService: DomainResolutionServiceType {
+    let server: RPCServer = .forResolvingEns
+
+    func resolveAddress(string value: String) -> Promise<BlockieAndAddressOrEnsResolution> {
+
+        func resolveBlockieImage(addr: AlphaWallet.Address) -> Promise<BlockieAndAddressOrEnsResolution> {
+            BlockiesGenerator()
+                .promise(address: addr)
+                .map { image -> BlockieAndAddressOrEnsResolution in
+                    return (image, .resolved(.address(addr)))
+                }.recover { _ -> Promise<BlockieAndAddressOrEnsResolution> in
+                    return .value((nil, .resolved(.address(addr))))
+                }
+        }
+
+        let getEnsAddressCoordinator = GetENSAddressCoordinator(server: server)
+        let unstoppableDomainsV1Resolver = UnstoppableDomainsV1Resolver(server: server)
+        let unstoppableDomainsV2Resolver = UnstoppableDomainsV2Resolver(server: server)
+
+        let services: [CachebleAddressResolutionServiceType] = [
+            getEnsAddressCoordinator,
+            unstoppableDomainsV1Resolver,
```

Comment:
Should v2 be before v1? (I don't know)

Reply:
actually i don't know, we can update the order.